sdks/wasm/binding_support.js

```diff
@@ -726,6 +728,38 @@ var BindingSupportLib = {
 				return BINDING.call_method (method, null, signature, arguments);
 			};
 		},
+		wasm_get_core_type: function (obj)
+		{
+			var coreType = this.call_method (this.get_core_type, null, "so", [ "WebAssembly.Core."+obj.constructor.name ]);
+			if (typeof coreType !== "undefined") {
+				BINDING.mono_wasm_core_map.set(obj.constructor, coreType);
+			}
+			return coreType;
+		},
+		core_object_type: function(obj) {
+			var coreType = BINDING.mono_wasm_core_map.get(obj.constructor);
+			if (typeof coreType === "undefined") {
+				var switchName = obj.constructor.name;
+				switch (true) {
+					case switchName === "Array":
+					case switchName === "ArrayBuffer":
+					case switchName === "Int8Array":
+					case switchName === "Uint8Array":
+					case switchName === "Uint8ClampedArray":
+					case switchName === "Int16Array":
+					case switchName === "Uint16Array":
+					case switchName === "Int32Array":
+					case switchName === "Uint32Array":
+					case switchName === "Float32Array":
+					case switchName === "Float64Array":
+					case switchName === "Function":
+					case switchName === "SharedArrayBuffer":
+						coreType = this.wasm_get_core_type(obj);
+						break;
+				}
+		  }
```

Comment:
nit: indentation

Reply:
is there a specific reason why the switch block is written this way? it is by far not how things are typically written in JS

Reply:
Arghh, absolutely correct I did not go back and change it correctly.  It was a leftover from a switch for instanceof which did not work reliably.
Addressed in recent commit.